web-nfc/resources/nfc-helpers.js

```diff
@@ -192,7 +192,12 @@ function assertWebNDEFMessagesEqual(message, expectedMessage) {
     assert_equals(record.recordType, expectedRecord.recordType);
     assert_equals(record.mediaType, expectedRecord.mediaType);
     assert_equals(record.id, expectedRecord.id);
-
+    if (record.recordType == 'text') {
+      assert_equals(record.encoding == null? 'utf-8': record.encoding,
```

Comment:
In those call sites of this function, usually |record| is from the mock nfc where its `encoding `and `lang `have already been set to non-null, |expectedRecord| is constructed from a `NDEFRecordInit`, means its `encoding `and `lang `are already set to non-null by ndef_record.cc, so I feel we can write like:
`assert_equals(record.encoding, expectedRecord.encoding);`
`assert_equals(record.lang, expectedRecord.lang);`
WDYT?

Reply:
Moreover, in case of types other than 'text', `encoding` and `lang` should be null for both records, so we can omit the check at line195 completely.